api/src/services/import-export.ts

```diff
@@ -341,6 +341,7 @@ export class ExportService {
 			const parser = new CSVParser({
 				transforms: [CSVTransforms.flatten({ separator: '.' })],
 				header: options?.includeHeader !== false,
+				...(input.length > 0 ? {} : { fields: [] }), // prevent CSVParser from erroring (issue #16719)
```

Comment:
Should we just short-circuit the export before even creating the parser object?

Reply:
my first thought was that optimally CSV with limit=0 should return the header without data but couldnt get that done in a time-boxed fashion.
Do you mean not giving a download when limit is set to `0` for any data type? I stuck with this way for consistency as the CSV is the only one erroring, JSON and XML already returned an empty file without erroring